wordpress-mobile/WordPress-FluxC-Android
# Fetching an empty list of posts should mark canLoadMore posts as false

Fixes #1481. Although #1481 is an API issue, which I'll report soon, there is an easy way for us to fix the issue in the client side. We simply set `canLoadMore` to `false` if we get an empty response. I think that's an OK thing to do by itself, so even if the API is fixed, I wouldn't mind having this in the code as a fail-safe for other API issues we might be unaware of.

**To test:**
Perform the tests outlined by @malinajirka in #1481 in WPAndroid for `issue/fix-can-load-more-for-fetching-post-list-2` [branch](https://github.com/wordpress-mobile/WordPress-Android/tree/issue/fix-can-load-more-for-fetching-post-list-2):

```
1. Login and switch to a site where you are in the contributor role (*I think a couple of other conditions need to be met)
2. Open PostList
3. Click on Search
4. Type [search_query]
5. Scroll down
6. Notice the list keeps fetching more items indefinitely
```

@malinajirka I don't currently have a site setup to test this, do you mind checking if you can reproduce the issue in this branch? Thanks!
